### PR TITLE
detect existing VRViewer in scene.

### PR DESCRIPTION
**Pull request by DuFF14**
This PR moves the main rendering loop from DisplayController to VRViewer. The biggest change is that a VRViewer with a camera (tagged "MainCamera") now exists in the scene before runtime (but doesn't have to). This plays nicely with Unity's VR setup, as Unity VR projects will always have a MainCamera in the scene, likely referenced by other objects in the scene. A Unity VR scene can be converted to an OSVR-Unity VR scene simply by adding a VRViewer component to the MainCamera, without breaking existing references.

Changes here also include the ill-advised, expensive "DirectMode Preview" by way of leaving the VRViewer camera on during rendering. If performance suffers in DirectMode, the first optimization to make is turning this option off. It is configurable via a boolean in the editor on the DisplayController component.

These changes are the result of integrating OSVR into Unity's new VR Samples project. They are designed to make turning a Unity VR project into an OSVR-Unity VR project as simple as possible.

### CODE REVIEW

File: OSVR-Unity/Assets/OSVRUnity/src/DisplayController.cs [hunk 1]
```diff
@@ -266,25 +235,66 @@ private void CreateHeadAndEyes()
...
+                        // create Viewer's VREyes
+                        uint eyeCount = (uint)_displayConfig.GetNumEyesForViewer(viewerIndex); //get the number of eyes for this viewer
+                        viewer.CreateEyes(eyeCount);
+                    }
+                }
+
                 // loop through viewers because at some point we could support multiple viewers
                 // but this implementation currently supports exactly one
-                for (uint viewerIndex = 0; viewerIndex < _viewerCount; viewerIndex++)
+                for (; viewerIndex < _viewerCount; viewerIndex++)
```
Comment (JeroMiya) [inline OSVR-Unity/Assets/OSVRUnity/src/DisplayController.cs]: Was this change intentional? it looks like ```viewerIndex``` is always equal to ```viewersInScene.Length``` at the start of this loop.

**Comment by JeroMiya** [comment]
See comment on ```viewerIndex``` but otherwise this looks good on a sight reading.

File: OSVR-Unity/Assets/OSVRUnity/src/DisplayController.cs (hunk 1 above)
Comment (DuFF14) [inline OSVR-Unity/Assets/OSVRUnity/src/DisplayController.cs]: Correct. Maybe I should be more explicit about requiring exactly one viewer in the scene? 
If it finds a VRViewer exists in the scene already, it doesn't need to enter the second loop. 
If it doesn't find a VRViewer in the scene, viewerIndex is still at 0, and it enters the second loop to create "_viewerCount" VRViewers, which should always be 1 for now.

File: OSVR-Unity/Assets/OSVRUnity/src/VRViewer.cs [hunk 2]
```diff
@@ -117,7 +158,92 @@ public void UpdateEyes()
...
+                        _disabledCamera = false;
+                    }
+                    yield return new WaitForEndOfFrame();
+                    if (DisplayController.UseRenderManager && DisplayController.CheckDisplayStartup())
+                    {
+                        // Issue a RenderEvent, which copies Unity RenderTextures to RenderManager buffers
+#if UNITY_5_2 || UNITY_5_3
+                        GL.IssuePluginEvent(DisplayController.RenderManager.GetRenderEventFunction(), OsvrRenderManager.RENDER_EVENT);
+#else
+                        Debug.LogError("GL.IssuePluginEvent failed. This version of Unity is not supported by RenderManager.");
```
Comment (rpavlik) [inline OSVR-Unity/Assets/OSVRUnity/src/VRViewer.cs]: can we just issue this error once and then flip the UseRenderManager flag to false so they don't flood it

Comment (DuFF14) [inline OSVR-Unity/Assets/OSVRUnity/src/VRViewer.cs]: Sounds reasonable. I will make that change.
